parquet/src/main/java/org/apache/iceberg/parquet/ReadConf.java

```diff
@@ -157,6 +164,23 @@ ParquetFileReader reader() {
     return shouldSkip;
   }
 
+  private Map<Long, Long> generateRowGroupsStartRowPos() {
```

Comment:
Why does this separately read the Parquet file to create a map that is used to initialize an array, when the starting position could be set for the array in the existing loop? I don't think this method is needed.

Reply:
The existing loop of row groups is based on the row group set that had been filtered with `options`.  So we need to read the Parquet file without any filter to get each starting row position of row group.

Reply:
You're right. Good catch!

Can you add some comments to explain why this is needed for later?

Reply:
Done.